### PR TITLE
Add several R and Bioconductor recipes

### PR DESCRIPTION
**Pull request by sebastian-luna-valero**
* [X] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [X] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Adds:
* https://bioconductor.org/packages/release/bioc/html/BiSeq.html
* http://bioconductor.org/packages/release/bioc/html/ChIPQC.html
* https://www.bioconductor.org/packages/release/bioc/html/gcrma.html
* https://bioconductor.riken.jp/packages/3.1/bioc/html/simpleaffy.html
* https://cran.r-project.org/web/packages/HiddenMarkov/index.html
* https://cran.r-project.org/web/packages/pvclust/index.html
* https://cran.r-project.org/web/packages/sciplot/index.html

### CODE REVIEW

**Comment by sebastian-luna-valero** [comment]
Adding also missing dependencies:
* https://bioconductor.org/packages/release/bioc/html/globaltest.html
* https://cran.r-project.org/web/packages/lokern/index.html
* https://cran.r-project.org/web/packages/betareg/index.html
* https://bioconductor.org/packages/release/bioc/html/chipseq.html
* https://bioconductor.org/packages/release/data/annotation/html/TxDb.Celegans.UCSC.ce6.ensGene.html
* https://bioconductor.org/packages/release/data/annotation/html/TxDb.Hsapiens.UCSC.hg18.knownGene.html
* https://bioconductor.org/packages/release/data/annotation/html/TxDb.Hsapiens.UCSC.hg38.knownGene.html
* http://bioconductor.org/packages/release/data/annotation/html/TxDb.Mmusculus.UCSC.mm10.knownGene.html
* http://bioconductor.org/packages/release/data/annotation/html/TxDb.Mmusculus.UCSC.mm9.knownGene.html
* https://bioconductor.org/packages/release/data/annotation/html/TxDb.Rnorvegicus.UCSC.rn4.ensGene.html

**Comment by sebastian-luna-valero** [comment]
Adding also:
* https://cran.r-project.org/web/packages/sfsmisc/index.html

File: recipes/bioconductor-simpleaffy/build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
```
Comment (bgruening) [inline recipes/bioconductor-simpleaffy/build.sh]: @sebastian-luna-valero can you fix the build.sh files and remove the empty line?

File: recipes/r-hiddenmarkov/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{% set version = '1.8-8' %}
```
Comment (bgruening) [inline recipes/r-hiddenmarkov/meta.yaml]: conda-forge?

File: recipes/r-hiddenmarkov/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  home: ftp://ftp.gns.cri.nz/pub/davidh/sslib/r-repo/
+  license: GPL (>= 2)
+  summary: Contains functions for the analysis of Discrete Time Hidden Markov Models, Markov
+    Modulated GLMs and the Markov Modulated Poisson Process. It includes functions for
+    simulation, parameter estimation, and the Viterbi algorithm. See the topic "HiddenMarkov"
+    for an introduction to the package, and "Change Log" for a list of recent changes.
+    The algorithms are based of those of Walter Zucchini.
+  license_family: GPL3
+
+# The original CRAN metadata for this package was:
```
Comment (bgruening) [inline recipes/r-hiddenmarkov/meta.yaml]: Comments removal?

File: recipes/r-lokern/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,57 @@
+{% set version = '1.1-8' %}
+
```
Comment (bgruening) [inline recipes/r-lokern/meta.yaml]: conda-foreg?

File: recipes/r-pvclust/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+{% set version = '2.0-0' %}
```
Comment (bgruening) [inline recipes/r-pvclust/meta.yaml]: conda-forge as well?

File: recipes/bioconductor-simpleaffy/build.sh (hunk 1 above)
Comment (sebastian-luna-valero) [inline recipes/bioconductor-simpleaffy/build.sh]: thanks @bgruening 

FYI: these empty lines are added by `scripts/bioconductor_skeleton.py` so should we fix that there as well?

Comment (bgruening) [inline recipes/bioconductor-simpleaffy/build.sh]: Yes, this would be great!

**Comment by sebastian-luna-valero** [comment]
FYI @bgruening I removed the R recipes and I am still getting `no space left on device`:
https://travis-ci.org/bioconda/bioconda-recipes/jobs/262148259#L1211

Happy to wait for your input.

**Comment by bgruening** [comment]
@sebastian-luna-valero but I guess your genome packages are several GB big?

**Comment by sebastian-luna-valero** [comment]
This is what I see:
```
BiSeq_1.16.0.tar.gz --> 872K
ChIPQC_1.12.2.tar.gz --> 1.4M
TxDb.Celegans.UCSC.ce6.ensGene_3.2.2.tar.gz --> 6.7M
TxDb.Hsapiens.UCSC.hg18.knownGene_3.2.2.tar.gz --> 16M
TxDb.Hsapiens.UCSC.hg38.knownGene_3.4.0.tar.gz --> 31M
TxDb.Mmusculus.UCSC.mm10.knownGene_3.4.0.tar.gz --> 14M
TxDb.Mmusculus.UCSC.mm9.knownGene_3.2.2.tar.gz --> 13M
TxDb.Rnorvegicus.UCSC.rn4.ensGene_3.2.2.tar.gz --> 11M
chipseq_1.26.0.tar.gz --> 2.2M
gcrma_2.48.0.tar.gz --> 171K
globaltest_5.30.0.tar.gz --> 535K
simpleaffy_2.52.0.tar.gz --> 618K
```

96 MB in total

**Comment by roryk** [comment]
Would love to have this included-- is there something I can help with?

**Comment by daler** [comment]
Thanks @roryk, we need to generalize https://github.com/bioconda/bioconda-recipes/pull/5485 into a skeleton script to create data packages with post-link downloading, which avoids consuming limited space on the channel.

To help with the skeleton script, if you could figure out a good way to decide whether or not to use the post-link strategy based on the the contents of a bioconductor tarball, that would help. For example, can we just get away with a size check? But even if the tarball is large, if code needs to be compiled then we should not use the post-link strategy, so maybe we need to also check for C/C++/Fortran source files. I don't know what other kinds of things we need to worry about, so figuring out the lay of the land on these data-heavy packages would be helpful.

In the short term, you can try copy-pasting-modifying #5485 for the TxDb packages, since I'm pretty sure they should use post-link strategy.

**Comment by sebastian-luna-valero** [comment]
@daler many thanks for adding the bioconductor data packages in https://github.com/bioconda/bioconda-recipes/pull/5854. Has the `post-link` process been automated into `scripts/bioconductor/bioconductor_skeleton.py`? or do we need to update the docs to point to these recipes as examples of what needs to be done in order to upload them properly to bioconda?

**Comment by daler** [comment]
@sebastian-luna-valero as long as you're using the latest bioconda-utils, this happens automatically. Be sure to use `bioconda-utils bioconductor-skeleton`, not the one in bioconda-recipes' `scripts` dir.

**Comment by daler** [comment]
(now that bioconda-utils bioconductor-skeleton seems to be working well,  I will remove bioconda-recipes `scripts/bioconductor/bioconductor_skeleton.py` to avoid confusion)

**Comment by sebastian-luna-valero** [comment]
Great @daler , should I update the docs here http://bioconda.github.io/guidelines.html#r-bioconductor ? or do you prefer to add it yourself?

**Comment by daler** [comment]
That would be great, thank you. I had forgotten to update the docs.

**Comment by daler** [comment]
xref: https://github.com/bioconda/bioconda-recipes/pull/5899

**Comment by sebastian-luna-valero** [comment]
As requested: https://github.com/bioconda/bioconda.github.io/pull/9

**Comment by sebastian-luna-valero** [comment]
Waiting for the following PRs to be merged over in conda-forge:
* https://github.com/conda-forge/staged-recipes/pull/3446
* https://github.com/conda-forge/staged-recipes/pull/3447
* https://github.com/conda-forge/staged-recipes/pull/3448
* https://github.com/conda-forge/staged-recipes/pull/3449
* https://github.com/conda-forge/staged-recipes/pull/3450

**Comment by roryk** [comment]
Thanks so much for putting all this together!

**Comment by bgruening** [comment]
Sebastian r-sfsmisc should also move to conda-forge isn't it?

**Comment by sebastian-luna-valero** [comment]
Thanks for the heads up @bgruening 

Indeed, `r-sfsmisc` is being added in https://github.com/conda-forge/staged-recipes/pull/3449

**Comment by sebastian-luna-valero** [comment]
After adding the R packages over at `conda-forge`:
https://github.com/conda-forge/staged-recipes/pull/3872

I am finally able to add these Bioconductor packages to Bioconda.

**Comment by roryk** [comment]
Awesome! Thank you so much.

**Pull request merged**